### PR TITLE
Add RAVL tree for Coarse Provider

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Add RAVL tree for Coarse Provider.
This code is a part of from https://github.com/oneapi-src/unified-memory-framework/pull/378.
The code of RAVL tree comes from https://github.com/pmem/pmdk/blob/master/src/core/ravl.c

Ref: #378 

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: src/ravl/ravl.c [hunk 1]
```diff
@@ -0,0 +1,551 @@
+/*
+ *
+ * Copyright (C) 2018-2024 Intel Corporation
```
Comment (lukaszstolarczuk) [inline src/ravl/ravl.c:3]: why `2018-`?

File: src/ravl/ravl.c [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define RAVL_DEFAULT_DATA_SIZE (sizeof(void *))
+
+enum ravl_slot_type {
+    RAVL_LEFT,
+    RAVL_RIGHT,
+
```
Comment (lukaszstolarczuk) [inline src/ravl/ravl.c:30]: are these empty lines here relevant somehow?

File: src/ravl/ravl.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ * Copyright (C) 2018-2023 Intel Corporation
```
Comment (lukaszstolarczuk) [inline src/ravl/ravl.h:3]: `2024`

**Review by bratpiorka — APPROVED** [review]

File: src/ravl/ravl.c (hunk 1 above)
Comment (ldorau) [reply src/ravl/ravl.c:3]: The code of RAVL tree comes from https://github.com/pmem/pmdk/blob/master/src/core/ravl.c

File: src/ravl/ravl.h (hunk 3 above)
Comment (ldorau) [reply src/ravl/ravl.h]: Done

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**